prow/tide/tide.go

```diff
@@ -834,75 +825,102 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
 				}
 			}
 		}
-		for retry := 0; retry < maxRetries; retry++ {
-			if err := c.ghc.Merge(sp.org, sp.repo, int(pr.Number), github.MergeDetails{
-				SHA:         string(pr.HeadRefOID),
-				MergeMethod: string(mergeMethod),
-			}); err != nil {
-				// TODO: Add a config option to abort batches if a PR in the batch
-				// cannot be merged for any reason. This would skip merging
-				// not just the changed PR, but also the other PRs in the batch.
-				// This shouldn't be the default behavior as merging batches is high
-				// priority and this is unlikely to be problematic.
-				// Note: We would also need to be able to roll back any merges for the
-				// batch that were already successfully completed before the failure.
-				// Ref: https://github.com/kubernetes/test-infra/issues/10621
-				if _, ok := err.(github.ModifiedHeadError); ok {
-					// This is a possible source of incorrect behavior. If someone
-					// modifies their PR as we try to merge it in a batch then we
-					// end up in an untested state. This is unlikely to cause any
-					// real problems.
-					log.WithError(err).Warning("Merge failed: PR was modified.")
-					break
-				} else if _, ok = err.(github.UnmergablePRBaseChangedError); ok {
-					// Github complained that the base branch was modified. This is a
-					// strange error because the API doesn't even allow the request to
-					// specify the base branch sha, only the head sha.
-					// We suspect that github is complaining because we are making the
-					// merge requests too rapidly and it cannot recompute mergability
-					// in time. https://github.com/kubernetes/test-infra/issues/5171
-					// We handle this by sleeping for a few seconds before trying to
-					// merge again.
-					log.WithError(err).Warning("Merge failed: Base branch was modified.")
-					if retry+1 < maxRetries {
-						time.Sleep(backoff)
-						backoff *= 2
+		// mergePR atempts 1 merge and returns a bool indicating if we should try
+		// to merge the remaining PRs and possibly an error.
+		mergePR := func() (bool, error) {
+			var err error
+			for retry := 0; retry < maxRetries; retry++ {
+				if err = c.ghc.Merge(sp.org, sp.repo, int(pr.Number), github.MergeDetails{
+					SHA:         string(pr.HeadRefOID),
+					MergeMethod: string(mergeMethod),
+				}); err != nil {
```

Comment:
Can you minimize the indenting here? How about, for example:

```golang
err := c.g.M(o,r,n)
if err == nil {
  return true, nil
}
// Excessive commenting
if _, ok := err.(g.MHE); ok {
  return true, fmt.Errorf("PR was modified: %v", err)
}
```

Reply:
Done.